xmodule/tests/test_video.py

```diff
@@ -731,7 +730,7 @@ def test_export_to_xml(self, mock_val_api):
             video_id=edx_video_id,
             static_dir=EXPORT_IMPORT_STATIC_DIR,
             resource_fs=self.file_system,
-            course_id=str(self.descriptor.runtime.course_id.for_branch(None)),
+            course_id=self.descriptor.scope_ids.usage_id.context_key,
```

Comment:
Why was `for_branch(None)` done here before? CourseKeys can't have versions, can they?

Reply:
Oh, hey, I guess they can have both branch and version identifiers:

https://github.com/openedx/opaque-keys/blob/c35470ead6d7f4855a45e400555dd3d1a14a08f1/opaque_keys/edx/locator.py#L354-L365

So is there a risk that we'll end up with a course_id with version information here, where before it had none?

Reply:
There is one weird edge case that I know of where course keys can have version information attached to them, and that happens when you're reading the UsageKey of an item that has just been modified during a Studio publish, while you're in the same process. It's probably a bug. I wrote a comment about it here:

https://github.com/openedx/edx-platform/blob/master/cms/djangoapps/contentstore/outlines.py#L25-L43

I don't know if it's something you really have to worry about in this scenario, but I just wanted to make sure you were aware.

Reply:
@ormsbee, that's good to know. I added a custom `init` to `XModuleToXBlockMixin` to verify this, and indeed there are some cases when the `context_key` contains a branch and version - e.g. it happens when we're submitting an answer to a ProblemBlock. While viewing a unit page, it happens for _some_ instances of `SequenceBlock`, `SectionBlock`, and `CourseBlock`. However, the old `runtime.course_id` contained this information too in these scenarios, so we're not adding more data.

Reply:
@Agrendalath: Okay, as long as we're consistent with existing behavior. Thank you for investigating.